SIP + Master merge

jkim931 opened the pull request:
SIP är mergeat med master nu! Går att ringa, inlogg är nu 7001 resp. 7002 igen, klarar av att synca osv.!
Denna pull request ersätter samuels SIP-call pullrequest (ersätter #31)

Commit:
Gör så man kan ringa sip-samtal. Användarna (som jag vet är jättebra att lägga upp på git) är 7001 psw: polisen1 samt 7002 psw: polisen2. Known issue: Pretty much everything. Fungerar typ alltid att ringa ut men väldigt sällan att ta emot samtalet.

Commit:
Blev en liten bugg att appen hängde sig när man skapade ärenden. Detta är nu åtgärdat.

jkim931 committed:
Merge branch 'call' of https://github.com/samni698/AinAppen into review

SIP nu mergeat med Master! w00p

Conflicts:
	res/values/strings.xml
	src/gov/polisen/ainappen/AddCaseFragment.java
	src/gov/polisen/ainappen/GlobalData.java
	src/gov/polisen/ainappen/LoginAcitivity.java
	src/gov/polisen/ainappen/MainActivity.java

jkim931 committed:
Merge branch 'master' of https://github.com/Rovanion/AinAppen into review

willedanielsson commented:
DET FUNKAR! JÄÄÄTTE BRA!

JakobBacklund commented:
<uses-sdk
        android:minSdkVersion="14"
        android:targetSdkVersion="19" />

Står två gånger i AndroidManifest och skapar problem vid inloggning.

Rovanion commented:
@JakobBacklund Jag såg det i manifestet igår fast i en annan branch. Lika bra att vi ordnar det i det här pull requestet som i en annan.

jkim931 committed:
IMPORTANTE: Handles server dbsync timeout!

Fixat så att användaren får ett meddelande om nätverk helt saknas, eller
nätverket är för dålig (10 sec timeout på synctråden)!

jkim931 committed:
Fixed small warnings and XML-duplicate rows

fixat comment + warnings

Rovanion commented:
Crash med följande stacktrace:

``` log
05-07 11:29:52.573: E/AndroidRuntime(2949): FATAL EXCEPTION: main
05-07 11:29:52.573: E/AndroidRuntime(2949): java.lang.NullPointerException
05-07 11:29:52.573: E/AndroidRuntime(2949):     at gov.polisen.ainappen.CallFragment.onCreateView(CallFragment.java:43)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:829)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1035)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at android.app.BackStackRecord.run(BackStackRecord.java:635)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1401)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at android.app.FragmentManagerImpl$1.run(FragmentManager.java:426)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at android.os.Handler.handleCallback(Handler.java:615)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at android.os.Handler.dispatchMessage(Handler.java:92)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at android.os.Looper.loop(Looper.java:137)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at android.app.ActivityThread.main(ActivityThread.java:4895)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at java.lang.reflect.Method.invokeNative(Native Method)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at java.lang.reflect.Method.invoke(Method.java:511)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:994)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:761)
05-07 11:29:52.573: E/AndroidRuntime(2949):     at dalvik.system.NativeStart.main(Native Method)
```

Rovanion commented:
Hittade också det här intressanta:

```
05-07 11:30:23.745: D/registering on sip server(2949): registering on sip server
05-07 11:30:25.237: D/registration on sip server failed(2949): registration on sip server failed
```

Så kraschen kanske beror på att registreringen misslyckades?

Rovanion commented:
Var också tvungen att flytta på <uses-sdk> till början av filen. Någon får testa för att se om det bara var Eclipse som jävlades med mig. Annars finns den commiten pushad till rovanion/sip eller så kan man applicera den här patchen:

``` diff
index 75ceba0..5724c2a 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -2,6 +2,10 @@
     package="gov.polisen.ainappen"
     android:versionCode="1"
     android:versionName="1.0" >
+    
+    <uses-sdk
+        android:minSdkVersion="14"
+        android:targetSdkVersion="19" />

     <uses-permission android:name="android.permission.WRITE_EX
     </uses-permission>
@@ -40,10 +44,6 @@
     <uses-permission android:name="android.permission.WRITE_EX
     <uses-permission android:name="android.permission.INTERNET

-    <uses-sdk
-        android:minSdkVersion="14"
-        android:targetSdkVersion="19" />
-
     <uses-permission android:name="android.permission.ACCESS_N
     </uses-permission>

```
